text/chapter5.md

```diff
@@ -426,66 +381,80 @@ So, constructing values of algebraic data types is simple, but how do we use the
 Let's see an example. Suppose we want to convert a `Shape` into a `String`. We have to use pattern matching to discover which constructor was used to construct the `Shape`. We can do this as follows:
 
 ```haskell
-showPoint :: Point -> String
-showPoint (Point { x: x, y: y }) =
-  "(" <> show x <> ", " <> show y <> ")"
+{{#include ../exercises/chapter5/src/Data/Picture.purs:showShape}}
 
-showShape :: Shape -> String
-showShape (Circle c r)      = ...
-showShape (Rectangle c w h) = ...
-showShape (Line start end)  = ...
-showShape (Text p text) = ...
+{{#include ../exercises/chapter5/src/Data/Picture.purs:showPoint}}
 ```
 
 Each constructor can be used as a pattern, and the arguments to the constructor can themselves be bound using patterns of their own. Consider the first case of `showShape`: if the `Shape` matches the `Circle` constructor, then we bring the arguments of `Circle` (center and radius) into scope using two variable patterns, `c` and `r`. The other cases are similar.
 
-`showPoint` is another example of pattern matching. In this case, there is only a single case, but we use a nested pattern to match the fields of the record contained inside the `Point` constructor.
+## Exercises
 
-## Record Puns
+1. (Easy) Write a function `circleAtOrigin` which constructs a `Circle` (of type `Shape`) centered at the origin with radius `10.0`.
+1. (Medium) Write a function `doubleScaleAndCenter` which scales the size of a `Shape` by a factor of `2.0` and centers it at the origin.
+1. (Medium) Write a function `shapeText` which extracts the text from a `Shape`. It should return `Maybe String`, and use the `Nothing` constructor if the input is not constructed using `Text`.
+
+## Newtypes
```

Comment:
We could use some exercises for this section. Also considering moving it to before the above exercise cluster. Follow-up proposed in #255